xdem/coreg/blockwise.py

```diff
@@ -17,645 +17,359 @@
-        coords[:, 1] = dem_arr.shape[0] * (1 - (coords[:, 1] - bounds.bottom) / (bounds.top - bounds.bottom))
-
-    # Generate a grid of x and y index coordinates.
-    grid_y, grid_x = np.mgrid[0 : dem_arr.shape[0], 0 : dem_arr.shape[1]]
-
-    if no_horizontal:
-        warped = dem_arr.copy()
-    else:
-        # Interpolate the sparse source-destination points to a grid.
-        # (row, col, 0) represents the destination y-coordinates of the pixels.
-        # (row, col, 1) represents the destination x-coordinates of the pixels.
-        new_indices = scipy.interpolate.griddata(
-            source_coords_scaled[:, [1, 0]],
-            destination_coords_scaled[:, [1, 0]],  # Coordinates should be in y/x (not x/y) for some reason..
-            (grid_y, grid_x),
-            method="linear",
+        coeff_y = self.robust_ransac(
+            self.x_coords,  # type: ignore
+            self.y_coords,  # type: ignore
+            self.shifts_y,  # type: ignore
+            threshold_ransac,
+            min_inliers_ransac,
+            max_iterations_ransac,
         )
-
-        # If the border should not be trimmed, just assign the original indices to the missing values.
-        if not trim_border:
-            missing_ys = np.isnan(new_indices[:, :, 0])
-            missing_xs = np.isnan(new_indices[:, :, 1])
-            new_indices[:, :, 0][missing_ys] = grid_y[missing_ys]
-            new_indices[:, :, 1][missing_xs] = grid_x[missing_xs]
-
-        order = {"nearest": 0, "linear": 1, "cubic": 3}
-
-        with warnings.catch_warnings():
-            # A skimage warning that will hopefully be fixed soon. (2021-06-08)
-            warnings.filterwarnings("ignore", message="Passing `np.nan` to mean no clipping in np.clip")
-            warped = skimage.transform.warp(
-                image=np.where(dem_mask, np.nan, dem_arr),
-                inverse_map=np.moveaxis(new_indices, 2, 0),
-                output_shape=dem_arr.shape,
-                preserve_range=True,
-                order=order[resampling],
-                cval=np.nan,
-            )
-            new_mask = (
-                skimage.transform.warp(
-                    image=dem_mask, inverse_map=np.moveaxis(new_indices, 2, 0), output_shape=dem_arr.shape, cval=False
-                )
-                > 0
+        if self.apply_z_correction:
+            coeff_z = self.robust_ransac(
+                self.x_coords,  # type: ignore
+                self.y_coords,  # type: ignore
+                self.shifts_z,  # type: ignore
+                threshold_ransac,
+                min_inliers_ransac,
+                max_iterations_ransac,
             )
+        else:
+            coeff_z = (0, 0, 0)
 
-        if dilate_mask:
-            new_mask = scipy.ndimage.binary_dilation(new_mask, iterations=order[resampling]).astype(new_mask.dtype)
+        self.mp_config.outfile = self.output_path_aligned
```

Comment:
I guess these two `self.mp_config.outfile=...` calls will make the operations fail when Multiprocessing is not used (`mp_config=None`).
Not sure how we make the behaviour consistent between the two cases... Maybe just adding `if mp_config is not None:` before the above call?